docs/interactionmanager.md

```diff
@@ -37,6 +37,157 @@ By default, queued tasks are executed together in a loop in one `setImmediate` b
 
 ---
 
+## Example
+
+### Basic
+
+```SnackPlayer name=InteractionManager%20Function%20Component%20Basic%20Example&supportedPlatforms=ios,android
+import * as React from 'react';
+import {
+  Alert,
+  Animated,
+  InteractionManager,
+  Platform,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+
+const instructions = Platform.select({
+  ios: 'Press Cmd+R to reload,\n' + 'Cmd+D or shake for dev menu',
+  android:
+    'Double tap R on your keyboard to reload,\n' +
+    'Shake or press menu button for dev menu',
+});
+
+const useMount = (func) => React.useEffect(() => func(), [])
+
+const useFadeIn = (duration = 5000) => {
+  const [opacity] = React.useState(new Animated.Value(0));
+
+  // Running the animation when the component is mounted
+  useMount(() => {
+    // Animated.timing() create a interaction handle by default, if you want to disabled that
+    // behaviour you can set isInteraction to false to disabled that.
+    Animated.timing(opacity, {
+      toValue: 1,
+      duration,
+    }).start();
+  });
+
+  return opacity;
+};
+
+function Ball({ onShown }) {
+  const opacity = useFadeIn();
+
+  // Running a method after the animation
+  useMount(() => {
+    InteractionManager.runAfterInteractions(() => onShown());
+  });
+
+  return <Animated.View style={[styles.ball, { opacity }]} />;
+}
+
+function App() {
+  const [show, setShow] = React.useState(false);
+
+  return (
+    <View style={styles.container}>
+      <Text>{instructions}</Text>
+      <Ball onShown={() => Alert.alert('Animation is done')} />
+    </View>
+  );
+}
+
+export default App;
+
+const styles = StyleSheet.create({
+  container: { flex: 1, justifyContent: 'center', alignItems: 'center' },
+  ball: {
+    width: 100,
+    height: 100,
+    backgroundColor: 'salmon',
+    borderRadius: 100,
+  },
+});
+```
+
+### Advanced
+
+```SnackPlayer name=InteractionManager%20Function%20Component%20Advanced%20Example&supportedPlatforms=ios,android
+import * as React from 'react';
+import {
+  Alert,
+  Animated,
+  InteractionManager,
+  Platform,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+
+const instructions = Platform.select({
+  ios: 'Press Cmd+R to reload,\n' + 'Cmd+D or shake for dev menu',
+  android:
+    'Double tap R on your keyboard to reload,\n' +
+    'Shake or press menu button for dev menu',
+});
+
+const useMount = (func) => React.useEffect(() => func(), [])
+
+// You can create a custom interaction/animation and add
+// support for InteractionManager
+const useCustomInteraction = (timeLocked = 2000) => {
+  useMount(() => {
+    const handle = InteractionManager.createInteractionHandle();
+
+    setTimeout(
+      () => InteractionManager.clearInteractionHandle(handle),
+      timeLocked
+    );
+
+    return () => InteractionManager.clearInteractionHandle(handle);
+  });
+};
+
+function Ball({ onInteractionIsDone }) {
+  useCustomInteraction()
```

Comment:
Also please run code through the Prettier (prettier.io). There are few missing semicolons at first glance.

Reply:
Sure

Reply:
@Simek some specific config for prettier ??